nemo/collections/common/parts/utils.py

```diff
@@ -105,3 +107,27 @@ def extend_instance(obj, mixin):
     obj.__class__ = type(
         base_cls_name, (mixin, base_cls), {}
     )  # mixin needs to go first for our forward() logic to work
+
+
+def mask_sequence_tensor(tensor: torch.Tensor, lengths: torch.Tensor):
+    """
+    For tensors containing sequences, zero out out-of-bound elements given lengths of every element in the batch.
+
+    tensor: tensor of shape (B, L), (B, D, L) or (B, D1, D2, L),
+    lengths: LongTensor of shape (B,)
+    """
+    batch_size, *_, max_lengths = tensor.shape
+
+    if len(tensor.shape) == 2:
```

Comment:
Could we re-use this function here to find the mask: https://github.com/NVIDIA/NeMo/blob/a7fbf6babe6d8a2c9f53b9e72f260151833297c1/nemo/collections/asr/models/aed_multitask_models.py#L68